functests/deployment_test.go

```diff
@@ -0,0 +1,41 @@
+package functests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("Deployment", func() {
+	When("Testing deployment", func() {
+		It("Should get the deployment, verify if the deployment is available, pods are running and service is available", func() {
+			k8sClient, err := getK8sClient()
+			Expect(k8sClient).NotTo(BeNil())
+			Expect(err).To(BeNil())
+
+			By("Verifying if " + operatorName + " deployment")
+			_, err = k8sClient.AppsV1().Deployments(operatorNamespace).Get(operatorName, metav1.GetOptions{})
+			Expect(err).To(BeNil())
+
+			By("Verifying if deployment " + operatorName + " is available")
+			err = WaitForDeployment(k8sClient, operatorName)
+			Expect(err).To(BeNil())
+
+			By("Verifying if pods are running")
+			labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"name": operatorName}}
+			listOptions := metav1.ListOptions{
+				LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+			}
+			pod, err := k8sClient.CoreV1().Pods(operatorNamespace).List(listOptions)
+			Expect(err).To(BeNil())
+			Expect(string(pod.Items[0].Status.Phase)).Should(Equal("Running"))
+			Expect(bool(pod.Items[0].Status.ContainerStatuses[0].Ready)).Should(Equal(true))
+
+			By("Verifying if " + operatorName + "-metrics service is available")
+			err = WaitForService(k8sClient, operatorName+"-metrics")
+			Expect(err).To(BeNil())
+		})
+	})
+})
```

Comment:
I don't think there is much value in having this test. BeforeEach() step in the other test should cover this already.

Reply:
Sorry, not saying to dump everything in one test. Just that deployment_test is redundant, as custom_resource_test already needs deployment to be up and running. Running deployment_test after custom_resource_test doesn't make any sense, and these tests run in random order.

Reply:
Agree that with the execution of test cases in random order, the situation when custom resource test runs prior to deployment test may look redundant but having a separate deployment test still helps in better readability and understanding which component should do what.

Reply:
I think it is also redundant.

Reply:
Done